Add basic toc generation for pandoc

pagdot opened the pull request:
As requested in #8152 

I've pretty much copied the toc parser code from asciidocext and modified it where necessary.

For toc generation with pandoc to work, I have to generate a standalone document. 

This lead to a few (solved) issues:

- added CSS causing issues: Solved by extracting `<body>` and ignoring everything else
- pandoc complains about missing title metadata: Solved by providing dummy title on commandline

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gohugoio/hugo?pullRequest=8911) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/gohugoio/hugo?pullRequest=8911) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gohugoio/hugo?pullRequest=8911) it.</sub>

MilkClouds commented:
Is this issue completed? Is there anything remaining to implement? Will it be merged?

bep commented:
Sorry for the delay, I'm not a Pandoc user and this has not been on top of my priority list.

The code looks good, but we need a test.

Since this PR was started I created a new integration test harness that allows integration tests to run full site builds from the package they live in (e.g. the `pandoc` package). Look for `hugolib.NewIntegrationTest` in the master branch.

nealkruis commented:
Eagerly waiting for this feature. @pagdot any updates?

pagdot commented:
rebased on the current master and added a very basic integration test. I'd like to verify that toc extractions work in a test, but don't really know how

pagdot commented:
@bep Does this look good to you or is still anything missing?
